tools/src/main/scala/scala/scalanative/codegen/CodeGen.scala

```diff
@@ -92,26 +87,32 @@ object CodeGen {
     private val deps             = mutable.Set.empty[Global]
     private val generated        = mutable.Set.empty[String]
     private val externSigMembers = mutable.Map.empty[Sig, Global.Member]
-    private val builder          = new ShowBuilder
-    import builder._
-
-    def gen(): ByteBuffer = {
-      genDefns(defns)
-      val body = builder.toString.getBytes("UTF-8")
-      builder.clear
-      genPrelude()
-      genConsts()
-      genDeps()
-      val prelude = builder.toString.getBytes("UTF-8")
-      val buffer  = ByteBuffer.allocate(prelude.length + body.length)
-      buffer.put(prelude)
-      buffer.put(body)
-    }
-
-    def genDeps() = deps.foreach { n =>
+
+    def gen(id: Option[Int], dir: VirtualDirectory): Unit = {
```

Comment:
Consider making `id` a `String` instead, and pass either `id.toString()` or `"out"` at call site.